gateway/src/inference/providers/anthropic.rs

```diff
@@ -448,6 +448,8 @@ struct AnthropicRequestBody<'a> {
     tool_choice: Option<AnthropicToolChoice<'a>>,
     #[serde(skip_serializing_if = "Option::is_none")]
     tools: Option<Vec<AnthropicTool<'a>>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    disable_parallel_tool_use: Option<bool>,
 }
```

Comment:
hey @hk1997 sorry I didn't catch this earlier but the tests did: `disable_parallel_tool_use` actually goes in the `AnthropicToolChoice` struct as a struct field, so 

```rust
/// We can instruct Anthropic to use a particular tool,
/// any tool (but to use one), or to use a tool if needed.
#[derive(Clone, Debug, PartialEq, Serialize)]
#[serde(tag = "type")]
#[serde(rename_all = "snake_case")]
enum AnthropicToolChoice<'a> {
    Auto { disable_parallel_tool_use: Option<bool> },
    Any { disable_parallel_tool_use: Option<bool> },
    Tool { name: &'a str, disable_parallel_tool_use: Option<bool> },
}
```

so we'll need to move this into there and then change the TryFrom to take a reference to the whole ToolConfig. 
Do you mind making that change?

Reply:
Glad to have good integration tests in place!

Yes I can make the change but can you help clarify the following:

Isn't the purpose of disable_parallel_tool_use is to enforce a sequential use of provided tools, and is not a localized behavior of a particular tool?

If I'm not wrong the curl request would look as follows:
```
curl https://api.anthropic.com/v1/messages \
  -H "x-api-key: YOUR_API_KEY" \
  -H "Content-Type: application/json" \
  -H "anthropic-version: 2023-06-01" \
  -d '{
    "model": "claude-3-opus-20240229",
    "messages": [
      {
        "role": "user",
        "content": "Find information about the Anthropics API using the web tool."
      }
    ],
    "tools": [
      {"name": "web"},
      {"name": "python"}
    ],
    "max_tokens": 500,
    "disable_parallel_tool_use": true
}'
```

Am I missing something?

Reply:
I don't disagree with this high-level reasoning.

However, 2 points:
*  the [api docs](https://docs.anthropic.com/en/api/messages#body-tool-choice) clearly show that the field goes in the tool choice block
* tool choice also has this property that this applies to the entire request. 
 
So doing a `TryFrom` on `&ToolCallConfig`  means you'll grab the tool choice from the `tool_choice` member and the `parallel_tool_choice` also and from there construct the enum above.
Does this make sense?

Reply:
Ahh yes, I missed the tool choice intent in the documentation.
Thank you for the clarification.

Yes it makes sense, I'll make the relevant changes.

Reply:
thank you!